package/MDAnalysis/analysis/encore/Ensemble.py

```diff
@@ -0,0 +1,273 @@
+        # and the analysis is therefore slower - we therefore use a
+        # single trajectory value when possible
+        if len(trajectory) == 1:
+            trajectory = trajectory[0]
+        MDAnalysis.Universe.__init__(self, topology, trajectory,
+                                     **kwargs)
+
+
+        if kwargs.get('format', None) != ArrayReader:
+
+            # Try to extract coordinates using Timeseries object
+            # This is significantly faster, but only implemented for certain
+            # trajectory file formats
+            try:
+                # frame_interval already takes into account
+                coordinates = self.universe.trajectory.timeseries(
+                    self.atoms, format='afc', skip=frame_interval)
+
+            # if the Timeseries extraction fails, fall back to a slower approach
+            except AttributeError:
+                coordinates = numpy.zeros(
+                    tuple([self.universe.trajectory.n_frames]) +
+                    self.atoms.coordinates().shape)
+
+                k = 0
+                for i, time_step in enumerate(self.universe.trajectory):
+                    if i%frame_interval == 0:
+                        coordinates[k] = self.atoms.coordinates(time_step)
+                        k+=1
+                coordinates = np.swapaxes(coordinates,0,1)
+
+            # Overwrite trajectory in universe with an ArrayReader
+            # object, to provide fast access and allow coordinates
+            # to be manipulated
+            self.trajectory = ArrayReader(coordinates)
+
+        self.topology_filename = topology
+
+
+    def get_coordinates(self, selection="", format='afc'):
+        """
+        Convenience method for extracting array of coordinates. In cases where
+        no selection is provided, this version is slightly faster than accessing
+        the coordinates through the timeseries interface (which always takes
+        a copy of the array).
+
+        Parameters
+        ----------
+
+            selection : str
+                Atom selection string in the MDAnalysis format.
+                 (see http://mdanalysis.googlecode.com/git/package/doc/html/documentation_pages/selections.html)
+
+            *format*
+               the order/shape of the return data array, corresponding
+               to (a)tom, (f)rame, (c)oordinates all six combinations
+               of 'a', 'f', 'c' are allowed ie "fac" - return array
+               where the shape is (frame, number of atoms,
+               coordinates)
+
+        """
+        if selection == "":
+            # If no selection is applied, return raw array
+            return self.trajectory.get_array(format=format)
+        else:
+            return self.trajectory.timeseries(self.select_atoms(selection),
+                                              format=format)
+
+
+    def align(self, selection="name CA", reference=None, weighted=True):
```

Comment:
I'm not 100% familiar with alignment.  Can this be done on the fly (frame by frame) or is it based on the average position over the entire trajectory?